ledger/participant-integration-api/src/main/scala/platform/apiserver/services/transaction/ApiTransactionService.scala

```diff
@@ -68,6 +71,7 @@ private[apiserver] object ApiTransactionService {
 private[apiserver] final class ApiTransactionService private (
     transactionsService: IndexTransactionsService,
     metrics: Metrics,
+    errorsVersionsSwitcher: ErrorCodesVersionSwitcher,
```

Comment:
What's the benefit of injecting a hard-coded if statement instead of a boolean flag? Do we anticipate outputting more than 2 types of error codes?

Reply:
I don't have strong opinion here.
A benefit of a separate class is its usages are easily searchable in an IDE. Might get helpful when e.g. we want to retire V1 error codes.